src/test/java/de/qabel/desktop/hockeyapp/VersionClientTest.java

```diff
@@ -0,0 +1,200 @@
+package de.qabel.desktop.hockeyapp;
+
+import de.qabel.core.accounting.CloseableHttpClientStub;
+import de.qabel.core.accounting.CloseableHttpResponseStub;
+import org.apache.http.NameValuePair;
+import org.apache.http.entity.BasicHttpEntity;
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONException;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+
+public class VersionClientTest {
+
+    private static final int VERSION_ID_1_1 = 208;
+    private static final int VERSION_ID_1_0 = 195;
+
+    CloseableHttpClientStub httpClient = new CloseableHttpClientStub();
+    private HockeyAppConfiguration config = new HockeyAppConfiguration("1.1", httpClient);
+    private VersionClient client = new VersionClient(config, httpClient);
+
+    @Test
+    public void checkAppVersion() {
+        assertEquals("1.1", config.getAppVersion());
+    }
+
+    @Test
+    public void findVersion() throws VersionNotFoundException, IOException {
+
+        String shortVersion = "1.1";
+        buildTestVersions();
+        HockeyAppVersion version = client.getVersion();
+
+        assertEquals(shortVersion, version.getShortVersion());
+    }
+
+    @Test(expected = IOException.class)
+    public void parseInvalidVersionCreateResponse() throws IOException {
+        String responseContent = "nbzuhbggzubzug";
+        client.parseVersionCreateResponse(responseContent);
+        client.createVersion("invalid");
+    }
+
+    @Test(expected = IOException.class)
+    public void parseInvalidVersionsResponse() throws IOException {
+        String responseContent = "nbzuhbggzubzug";
+        client.parseVersionsResponse(responseContent);
+    }
+
+    @Test
+    public void parseVersionCreateResponse() throws IOException {
+
+        String versionsResponseContent = getVersionCreateResponseString(config.getAppVersion());
+
+        client.parseVersionCreateResponse(versionsResponseContent);
+
+        assertEquals(VERSION_ID_1_1, client.getVersion().getVersionId());
+    }
+
+    @Test
+    public void parseVersionsResponse() throws IOException {
+        String versionsResponseContent = getVersionsJsonString();
+        List<HockeyAppVersion> versions = client.parseVersionsResponse(versionsResponseContent);
+        assertEquals(2, versions.size());
```

Comment:
This test is (or has become) useless. If this is the only test for `parseVersionsResponse`, then that method is tested badly because you don't assert the content of those versions. a 
```JAVA
List<HockeyAppVersion> versions = new LinkedList();
versions.add(null);
versions.add(null);
return versions;
```
would be enough to make this test green.

I guess the parseVersionsResponse method is tested by more tests so just remove this test